src/notifications/settings/firebase.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  NotificationTopicType,
+  WalletNotificationSettings,
+} from '@/notifications/settings/types';
+import messaging from '@react-native-firebase/messaging';
+import { trackChangedNotificationSettings } from '@/notifications/analytics';
+import { NotificationTopic } from '@/notifications/settings/constants';
+import { logger } from '@/logger';
+
+/**
+ Firebase functions for subscribing/unsubscribing to topics.
+ */
+export const subscribeWalletToAllEnabledTopics = (
+  settings: WalletNotificationSettings,
+  chainId: number
+): Promise<void[]> => {
+  return Promise.all(
+    Object.entries(settings.topics).map(([topic, isEnabled]) => {
+      if (isEnabled) {
+        return subscribeWalletToSingleNotificationTopic(
+          settings.type,
+          chainId,
+          settings.address,
+          topic
+        );
+      } else {
+        return Promise.resolve();
+      }
+    })
+  );
+};
+
+export const unsubscribeWalletFromAllNotificationTopics = (
+  type: string,
+  chainId: number,
+  address: string
+): Promise<void[]> => {
+  return Promise.all(
+    Object.values(NotificationTopic).map(topic =>
+      unsubscribeWalletFromSingleNotificationTopic(
+        type,
+        chainId,
+        address,
+        topic
+      )
+    )
+  );
+};
+
+export const subscribeWalletToSingleNotificationTopic = (
+  type: string,
+  chainId: number,
+  address: string,
+  topic: NotificationTopicType
+): Promise<void> => {
+  logger.debug(
+    `Notifications: subscribing ${type}:${address} to [ ${topic.toUpperCase()} ]`
+  );
+  return messaging()
+    .subscribeToTopic(`${type}_${chainId}_${address.toLowerCase()}_${topic}`)
+    .then(() =>
+      trackChangedNotificationSettings(chainId, topic, type, 'subscribe')
+    );
+};
+
+export const unsubscribeWalletFromSingleNotificationTopic = async (
+  type: string,
+  chainId: number,
+  address: string,
+  topic: NotificationTopicType
+) => {
+  logger.debug(
```

Comment:
nit: would love to add debug context for these if possible so we can turn on and off, similar to what's done in analytics.. may be a lil hard with currently implementation but would be a nice add. 

100% shouldn't block dis

Reply:
I think I added the correct logger debug context